snorkel/map/spark.py

```diff
@@ -5,42 +5,21 @@
 from .core import Mapper
 
 
-class SparkMapper(Mapper):
-    """Base class for any `Mapper` that runs on PySpark `Row` objects.
+def _update_fields(x: Row, mapped_fields: FieldMap) -> Row:
+    # `pyspark.sql.Row` objects are not mutable, so need to
+    # reconstruct
+    all_fields = x.asDict()
+    all_fields.update(mapped_fields)
+    return Row(**all_fields)
 
-    See `snorkel.map.core.Mapper`.
 
-    Parameters
-    ----------
-    field_names
-        A map from attribute names of the incoming data points
-        to the input argument names of the `run` method. If None,
-        the parameter names in the function signature are used.
-    mapped_field_names
-        A map from output keys of the `run` method to attribute
-        names of the output data points. If None, the original
-        output keys are used.
-    memoize
-        Memoize mapper outputs?
+def make_spark_mapper(mapper: Mapper) -> Mapper:
```

Comment:
If the only thing you need to do to make a Mapper a SparkMapper is to change the logic of _update_fields, why can't we subclass it and just overwrite that method? (I trust there's a good reason--otherwise this PR wouldn't exist--but it's not obvious from review)

Reply:
See description. We want to write a Mapper/Preprocessor/etc. once, then be able to use it with Spark without having to make a new subclass.